shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -195,53 +196,69 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContext(
     v8::Context::Scope destination_scope(destination_context);
     {
       auto source_promise = v8::Local<v8::Promise>::Cast(value);
-      auto* proxied_promise = new gin_helper::Promise<v8::Local<v8::Value>>(
-          destination_context->GetIsolate());
+      std::shared_ptr<gin_helper::Promise<v8::Local<v8::Value>>>
+          proxied_promise(new gin_helper::Promise<v8::Local<v8::Value>>(
+              destination_context->GetIsolate()));
       v8::Local<v8::Promise> proxied_promise_handle =
           proxied_promise->GetHandle();
 
+      v8::Global<v8::Context> global_then_source_context(
+          source_context->GetIsolate(), source_context);
+      v8::Global<v8::Context> global_then_destination_context(
+          destination_context->GetIsolate(), destination_context);
```

Comment:
This is redundant with the promise's context right? Should we expose a reference to the promise helper's context?

Reply:
I don't think the promise should expose it's context, I'd rather the promise didn't hold a context at all 😆